File: CareConnect/CareConnect.Models/Requests/EmployeeInsertRequest.cs
using CareConnect.Models.Responses;
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;
using static System.Collections.Specialized.BitVector32;

namespace CareConnect.Models.Requests
{
    public class EmployeeInsertRequest
    {
        public DateTime HireDate { get; set; }

        public string JobTitle { get; set; } = string.Empty;

        [JsonIgnore]
        public bool Employed { get; set; } = true; 

        [JsonIgnore]
        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public UserInsertRequest User {  get; set; }

        public QualificationInsertRequest? Qualification { get; set; }
    }
}

File: UI/careconnect_mobile/lib/models/requests/employee_insert_request.dart
import 'package:careconnect_mobile/models/requests/qualification_insert_request.dart';
import 'package:careconnect_mobile/models/requests/user_insert_request.dart';

import 'package:json_annotation/json_annotation.dart';

part 'employee_insert_request.g.dart'; // Don't open or edit this

@JsonSerializable()
class EmployeeInsertRequest {
  final DateTime hireDate;
  final String jobTitle;
  final UserInsertRequest user;
  final QualificationInsertRequest? qualification;

  EmployeeInsertRequest({
    required this.hireDate,
    required this.jobTitle,
    required this.user,
    this.qualification,
  });

  factory EmployeeInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$EmployeeInsertRequestFromJson(json);

  Map<String, dynamic> toJson() => _$EmployeeInsertRequestToJson(this);
}
